src/hooks/screen/useMessageHook.ts

```diff
@@ -113,6 +113,23 @@ function useMessageHook(): UseMessageHook {
     }
   };
 
+  const onOpenMediaPreview = (medias, index, navigation) => {
+    if (medias.find((media) => media.type === 'video')) {
+      navigation.push('VideoViewer', {
+        title: 'Video',
+        url: medias[index].video_path
+      });
+    } else {
+      navigation.push('ImageViewer', {
+        title: 'Photo',
+        index,
+        images: medias
+          .filter((media) => media.type === 'image')
+          .map((media) => ({url: media.asset_url}))
+      });
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `onOpenMediaPreview` function assumes that if any media in the array is a video, then the media at the provided index will also be a video. This could lead to incorrect behavior if the media at the given index is not a video. I suggest checking the type of the media at the given index instead.
```diff
-    if (medias.find((media) => media.type === 'video')) {
+    if (medias[index].type === 'video') {
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->